packages/firestore/src/core/bundle_types.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This is a separate file because rollup (or typescript) treat it as circular
+// dependency when it was in `./bundle.ts`, which breaks bundles module build.
```

Comment:
```suggestion
// This is a separate file because rollup (or typescript) treat it as circular
// dependency when it was in `./bundle.ts`, which breaks the bundle's module
// build.
```